controlplane/pkg/nsmd/nsmd_crossconnect_client.go

```diff
@@ -148,7 +147,7 @@ func (client *NsmMonitorCrossConnectClient) startPeerMonitor(clientConnection *m
 		remotePeer.connections[clientConnection.ConnectionID] = clientConnection
 		return
 	}
-	ctx, cancel := context.WithCancel(context.Background())
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
This is not right, since ctx is client Request context if we came here for first time.